src/_pytest/reports.py

```diff
@@ -347,6 +347,10 @@ def from_item_and_call(cls, item: Item, call: "CallInfo[None]") -> "TestReport":
             elif isinstance(excinfo.value, skip.Exception):
                 outcome = "skipped"
                 r = excinfo._getreprcrash()
+                if r is None:
```

Comment:
This branch is not covered. I'll turn this into an assert, a ValueError isn't really appropriate here, and hopefully it really can't happen.